### PR TITLE
Add JSON schemas

### LINKED ISSUE #172: Document schema for all JSON file arguments
Please consider documenting what the schema is for all commands that take a JSON file.

While there are example JSON for some of them (partitions, white label, permissions), it would be ideal all JSON schema for each such argument was documented with a description of what each field in the JSON means, and what its possible values are.

### PR DESCRIPTION
**Pull request by will-v-pi**
Add JSON schemas for the JSON inputs used.

Currently the URLs point to the jsonschema branch - these should be changed to the develop branch once merged.

Fixes #172

### CODE REVIEW

File: README.md
Comment (lurch) [inline README.md:955]: It looks like this PR also moves `sample-wl.json` to `json/sample-wl.json` ? (and same comment for the other "sample" JSON files too)

File: json/default-pt.json
Comment (lurch) [inline json/default-pt.json:2]: Should we ensure that `rp2350` is in the schema-URL somewhere, in case future chips need different schema files?

File: json/schemas/otp-contents-schema.json
Comment (lurch) [inline json/schemas/otp-contents-schema.json:4]: "...of the RP2350 OTP" ?

File: json/schemas/partition-table-schema.json
Comment (lurch) [inline json/schemas/partition-table-schema.json:64]: If `permissions` is required, are the sub-properties within `permissions` also required?

Comment (will-v-pi) [reply json/schemas/partition-table-schema.json:64]: No, just an empty permissions will work fine
```
"permissions": {}
```

File: json/schemas/partition-table-schema.json
Comment (lurch) [inline json/schemas/partition-table-schema.json:117]: It looks like this `permissions` part of the schema is identical to the earlier `permissions` part? Might be worth using https://json-schema.org/understanding-json-schema/structuring#defs to avoid repetition.

File: json/schemas/whitelabel-schema.json
Comment (lurch) [inline json/schemas/whitelabel-schema.json:49]: nitpick: I guess either the `.` should be a `,`, or the `In` should be capitalised?

**Comment by lurch** [comment]
Should one of these schema files contain all the info from section 13.9 of the RP2350 datasheet, or is this the wrong place for that?

**Comment by will-v-pi** [comment]
> Should one of these schema files contain all the info from section 13.9 of the RP2350 datasheet, or is this the wrong place for that?

I think this is the wrong place for that - that section of the datasheet is probably the best place for that information, as `picotool` should just handle all the redundancy settings etc for the predefined values without issue when reading/writing OTP.

File: json/schemas/otp-contents-schema.json
Comment (will-v-pi) [reply json/schemas/otp-contents-schema.json:4]: I'll go with "... of the RP-series device OTP"

File: json/default-pt.json
Comment (will-v-pi) [reply json/default-pt.json:2]: I think the schema would hopefully stay the same for future chips - this isn't really a proper URL anyway, if we wanted to do it properly then we could add GitHub Pages to host the schemas, and then host schemas for each released version of picotool (in case they do change at a later date)

File: json/schemas/whitelabel-schema.json
Comment (will-v-pi) [reply json/schemas/whitelabel-schema.json]: Typo - fixed

File: json/default-pt.json
Comment (will-v-pi) [reply json/default-pt.json:2]: Actually, we probably don't need full on GitHub Pages for this - the `raw.githubusercontent` URLs have versioning, so you can just use URLs like this:
```
https://raw.githubusercontent.com/raspberrypi/picotool/master/json/schemas/partition-table-schema.json
https://raw.githubusercontent.com/raspberrypi/picotool/develop/json/schemas/partition-table-schema.json
https://raw.githubusercontent.com/raspberrypi/picotool/2.1.0/json/schemas/partition-table-schema.json
```

I think this is probably the simplest solution - you can stick to a specific version, or track `master` or `develop`

Comment (lurch) [reply json/default-pt.json:2]: > I think the schema would hopefully stay the same for future chips

Ahhh, fair enough. Is that likely to be true for **all** the schemas that you're adding in this PR?

File: json/schemas/whitelabel-schema.json
Comment (lurch) [inline json/schemas/whitelabel-schema.json:55]: If bit 7 must be 1, does that imply that 128 is the minimum value? And if bits 0-4 must be 0, then I guess 255 isn't the maximum value?

File: json/schemas/whitelabel-schema.json
Comment (lurch) [inline json/schemas/whitelabel-schema.json:58]: If bit 7 must be 1, then I guess the pattern here should use `{2}` rather than `{0,2}` ?

File: json/schemas/whitelabel-schema.json
Comment (lurch) [inline json/schemas/whitelabel-schema.json:52]: Should the regex here use `{1,2}` rather than `{0,2}`? Because it seems like `0x` wouldn't be a valid value?

File: json/schemas/whitelabel-schema.json
Comment (will-v-pi) [reply json/schemas/whitelabel-schema.json:55]: Fixed - the range is 128-224. Technically only the 4 values `0x80`, `0xA0`, `0xC0` and `0xE0` are allowed, so I've changed the regex to match that

File: json/schemas/whitelabel-schema.json
Comment (will-v-pi) [reply json/schemas/whitelabel-schema.json]: Yep - fixed

File: json/schemas/otp-schema.json
Comment (lurch) [inline json/schemas/otp-schema.json:20]: I guess `items` here could also be a `$ref`, but I'll let you decide if it's worth it or not.

File: json/default-pt.json
Comment (will-v-pi) [reply json/default-pt.json:2]: > > I think the schema would hopefully stay the same for future chips
> 
> Ahhh, fair enough. Is that likely to be true for **all** the schemas that you're adding in this PR?

The hope would be that they will all be backwards compatible at least - but as I said if they do change then you can pin to a specific version using the URL, without needing any special folder structure.

**Review by kilograham — APPROVED** [review]

**Comment by lurch** [comment]
@will-v-pi Don't forget to update the schema URLs before hitting the big green Merge button :smile:

**Pull request merged**
